tensorflow/core/framework/op_kernel.cc

```diff
@@ -1346,7 +1348,8 @@ Status FindKernelRegistration(
   // default kernel.
   if (*reg == nullptr &&
       !IsSymbolicExecutionDevice(device_type.type_string())) {
-    const string default_key = Key(node_op, DEVICE_DEFAULT, label);
+    const string default_key = Key(node_op, /*device_type*/ DEVICE_DEFAULT,
```

Comment:
I'm a little surprised that you did not need to change DeviceNameUtils::ParseFullName and friends, how do you parse this new format?
>>jzhoulon: I think `DeviceNameUtils::ParseFullName` is parsing front-end name , such as when user specify "with tf.device("/job:bar/task:0/device:gpu:0")", the  full name is "/job:bar/task:0/device:gpu:0", user no need to specify a subdevice name since only one can register successful with the same device type. subdevice type is mainly for resolving kernel registration conflict with first party devices and low-level specialization.

 but please use /*device_type=*/. 
>>ok, I will modify it.

Reply:
done